### ISSUE TITLE
#[callback_unwrap] doesn't work for unit type

### ISSUE THREAD

**Issue by fomotoshi** [near/near-sdk-rs#839]
This code causes deserialization error.

<pre>#<span class="error">[near_bindgen]</span>
 impl Contract {
 #<span class="error">[private]</span>
 pub fn on_some(
 &mut self,
 foo: Foo,
 #<span class="error">[callback_unwrap]</span> _bar: (),
 ) {
 ...</pre>

This code works like as expected

<pre>#<span class="error">[near_bindgen]</span>
 impl Contract {
 #<span class="error">[private]</span>
 pub fn on_some(
 &mut self,
 foo: Foo,
 #<span class="error">[callback_result]</span> bar: Result<(), PromiseError>,
 ) {
 bar.unwrap();
 ...</pre>

**Comment by austinabell** [comment]
So the reason this is the case is that `callback_result` handles `()` as a special case where it allows serialized bytes to be empty, but this was not added for `callback_unwrap`.

For now, someone should just unwrap in the code you posted that works.

cc @itegulov do you think it makes sense for this to have that special case handled for `callback_unwrap` too? The only concern I have is that it silently changes the semantics of that attribute, so feels like it has to come with a major version change.

**Comment by itegulov** [comment]
Yeah, missed this case when I was working on unit support for `callback_result`, thanks for reporting.

> do you think it makes sense for this to have that special case handled for `callback_unwrap` too?

I feel like we should stay consistent here and handle `()` as a special case for `callback_unwrap` too.

> The only concern I have is that it silently changes the semantics of that attribute, so feels like it has to come with a major version change.

Well, honestly I would argue that this is a bug, but I am also happy for this to come with a major version change.

**Comment by austinabell** [comment]
> Well, honestly I would argue that this is a bug, but I am also happy for this to come with a major version change.

I mean, it was the semantics of the API before we handled this edge case specifically for `callback_result`, there may be developers that expect `null` to be serialized as a result and fail if not, and we would be breaking their usage.
